concepts/modularization/transition.md

```diff
@@ -0,0 +1,30 @@
+You can test operator on its own by applying operator CRD and creating the instance (you can install operator on your test cluster or even run it locally). When your operator works you can build the image and publish it (using the recommended pipeline template) and create a `ModuleTemplate` that contains a default custom resource used to install the module and operator deployment. You create one module template per release channel you want to support. Having `ModuleTemplate` you can start testing your module with `kyma-operator` . To do that you need to install kyma-operator and its custom resources in your test cluster, create a `Kyma` resource and add your module in the spec. `Kyma-operator` will install your operator CRD, will deploy your operator and start module installation by creating default custom resource. When your operator changes the status.state field it should be propagated to the `Kyma` resource status. You can install `kyma-operator` and enable/disable modules using `kyma CLI` or `kyma Dashboard`.
+
+## Phase 2 - first module managed by `kyma-operator` integrated with KEB
+
+### Phase 2a - new clusters only (internal usage)
+
+`kyma-operator` is deployed in the control-plane and integrated with Kyma Environment Broker (KEB). Additional plan (preview) is created in KEB that creates Kyma clusters with kyma-operator only (no call to reconciler). This plan is the first playground to test KEB integration without affecting existing clusters and flows. The plan is tested with all environments (dev, stage, prod) to ensure we have all major integration issues resolved. The first module (frontrunner) can be submitted by the team and installed in the cluster. The minimal submission process is described and fully automated (manual steps are allowed only before submission). The minimal submission process looks like:
+- team verifies on their own if new module (or version) is product standard compliant 
+- team creates pull request to the control-plane that contains new/updated ModuleTemplate (by doing it team declares product standard compliance and functional correctness - can be checkbox in PR template)
+- automated test verifies if the module operator can be installed/upgraded without issues for all release channels (with default settings) - it is not functional test or integration test 
+- automated test verifies if all images installed by operator are declared in the module template, signed by approved CA, vulnerability free
```

Comment:
```suggestion
- automated test verifies if all images installed by operator are declared in the module template, signed by approved CA, vulnerability free
```
I checked with @clebs and we want to check if it makes sense bundling the ModuleTemplate Information into the existing built module. That way we have only one image and all the checks can be done at build time so we could skip this

Reply:
We have to discuss it. Example:
Eventing operator installs eventing-controller and eventing-publishing-proxy. These 2 components are not directly installed by the kyma-operator, but should be signed and vulnerability free.

Reply:
```suggestion
4. An automated test verifies if all images installed by operator are declared in the module template, signed by approved CA, and free of vulnerabilities.
```